packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -52,6 +59,10 @@ export class SqsQueue implements events.IRuleTarget {
    * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/resource-based-policies-eventbridge.html#sqs-permissions
    */
   public bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const role = this.props.role ?? singletonEventRole(rule);
```

Comment:
Why do we need to create a role when none has been provided? To ensure that we keep the current behavior, we should leave it undefined.

Reply:
You're right. I have addressed your comment.